src/bulkusbdevice.ts

```diff
@@ -85,7 +85,7 @@ export class BulkUsbSingleton {
         const newDevices = [];
         const ret = newList.map(newDevice => {
           const oldDevice = this._previousDevices.find(x => x.equals(newDevice));
-          if (oldDevice) {
+          if (oldDevice && oldDevice.serialNumber != 'Unknown serial') {
```

Comment:
Should we try to add a unit test for this. Probably something that is good to cover.

Reply:
I agree, but _listDevices is in charge of generating the newList as well, how can such a test operate?

Reply:
I think you can do something similar to this. https://github.com/Huddly/device-api-usb/blob/master/tests/bulkusbdeviceSingleton.spec.ts#L48